src/Fhir.Anonymizer.Shared.Core.UnitTest/Visitors/AnonymizationVisitorTests.cs

```diff
@@ -83,16 +84,70 @@ public void GivenAnEncryptRule_WhenProcess_NodeShouldBeEncrypted()
             patientNode.Accept(visitor);
             patientNode.RemoveNullChildren();
 
-            var patientAddress = patientNode.Select("Patient.address[0].city").FirstOrDefault();
+            var patientCity = patientNode.Select("Patient.address[0].city").FirstOrDefault();
             var key = Encoding.UTF8.GetBytes("1234567890123456");
-            var plainValue = EncryptUtility.DecryptTextFromBase64WithAes(patientAddress.Value.ToString(), key);
+            var plainValue = EncryptUtility.DecryptTextFromBase64WithAes(patientCity.Value.ToString(), key);
             Assert.Equal("patienttestcity1", plainValue);
 
             patient = patientNode.ToPoco<Patient>();
             Assert.Single(patient.Meta.Security);
             Assert.Contains(SecurityLabels.ENCRYPT.Code, patient.Meta.Security.Select(s => s.Code));
         }
 
+        [Fact]
+        public void GivenAPrimitiveSubstituteRule_WhenProcess_NodeShouldBeSubstituted()
+        {
+            AnonymizationFhirPathRule[] rules = new AnonymizationFhirPathRule[]
+            {
+                new AnonymizationFhirPathRule("Patient.address.city", "address.city", "Patient", "substitute", AnonymizerRuleType.FhirPathRule, "Patient.address.city",
+                    new Dictionary<string, object> { {"replaceWith", "ExampleCity2020" } })           
+            };
+
+            AnonymizationVisitor visitor = new AnonymizationVisitor(rules, CreateTestProcessors());
+
+            var patient = CreateTestPatient();
+            var patientNode = ElementNode.FromElement(patient.ToTypedElement());
+            var patientCity = patientNode.Select("Patient.address[0].city").FirstOrDefault();
+            Assert.Equal("patienttestcity1", patientCity.Value.ToString());
+            var patientCountry = patientNode.Select("Patient.address[0].country").FirstOrDefault();
+            Assert.Equal("patienttestcountry1", patientCountry.Value.ToString());
+
+            patientNode.Accept(visitor);
+            patientNode.RemoveNullChildren();
+
+            patientCity = patientNode.Select("Patient.address[0].city").FirstOrDefault();
+            Assert.Equal("ExampleCity2020", patientCity.Value.ToString());
+            patientCountry = patientNode.Select("Patient.address[0].country").FirstOrDefault();
+            Assert.Equal("patienttestcountry1", patientCountry.Value.ToString());
```

Comment:
Check for patient.contact.city is not changed maybe better?

Reply:
Thanks. Updated